ocs_ci/deployment/node.py

```diff
@@ -0,0 +1,109 @@
+"""
+A class to represent generic node (host) objects which are
+added/removed from the cluster on need basis
+"""
+import os
+import logging
+
+import pytest
+import yaml
+
+
+from ocs_ci.framework import config, merge_dict
+from ocs_ci.ocs import constants
+
+logger = logging.getLogger(__name__)
+
+
+class Node(object):
+    """
+    A base class for representing a node in a cluster
+    """
+    def __init__(self, node_t):
+        # hostname
+        self.name = None
+        self.platform = config.ENV_DATA['platform']  # AWS/VMWARE
+        self.deployment_t = config.ENV_DATA['deployment_type']  # ipi/upi
+        # RHCOS/RHEL, this will be set in child
+        self.node_t = node_t
+
+    def prepare_node(self):
+        raise NotImplementedError(
+            "prepare_node should be impelented in Child class"
+        )
+
+
+class AWSNode(Node):
+
+    def __init__(self, node_conf, node_t):
+        super(AWSNode, self).__init__(node_t)
+        # Config of this node object
+        self.conf = node_conf
+        # boto3 aws instance object
+        self.node_obj = None
+
+        # Avoiding circular dependency
+        from .node_utils import NodeUtils
+        self.util = NodeUtils()
+
+    def prepare_node(self):
```

Comment:
Even if it's simple, these 'wrapper functions' should have a docstring so it's clear what they do in our generated doc.